Classes/common/query/CDTQQuerySqlTranslator.m

```diff
@@ -495,4 +545,31 @@ + (CDTQSqlParts *)selectStatementForAndClause:(NSArray *)clause usingIndex:(NSSt
     return parts;
 }
 
++ (CDTQSqlParts *)selectStatementForTextClause:(NSObject *)clause usingIndex:(NSString *)indexName
+{
+    if (!clause) {
+        return nil;  // no query here
+    }
+    
+    if (!indexName) {
+        return nil;
+    }
+    
+    if (![clause isKindOfClass:[NSDictionary class]]) {
+        return nil; // should never get here as this would not pass normalization
```

Comment:
Since this should never happen, I'm wondering if we should just let it fail with doesn't respond to selector errors. Failing hard would make it easier to find the issue if someone breaks normalisation. So I'm thinking remove the check and leave a comment like:
```
Should never be anything but a dictionary, anything else should fail normalisation
```

Thoughts @alfinkel ?